tests/js/server/shell/shell-transactions-view-noncluster.js

```diff
@@ -0,0 +1,312 @@
+              SEARCH ANALYZER(doc.text IN TOKENS('the quick brown', 'myText'), 'myText') 
+              SORT TFIDF(doc) 
+              LIMIT 4 
+              RETURN doc`;
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test suite
+////////////////////////////////////////////////////////////////////////////////
+
+function TransactionsIResearchSuite() {
+  'use strict';
+  let c = null;
+  let view = null;
+
+  return {
+
+    setUpAll: function() {
+      analyzers.save(
+        "myText",
+        "text",
+        { locale: "en.UTF-8", stopwords: [ ] },
+        [ "frequency", "norm", "position" ]
+      );
+    },
+
+    ////////////////////////////////////////////////////////////////////////////////
+    /// @brief set up
+    ////////////////////////////////////////////////////////////////////////////////
+
+    setUp: function () {
+      db._drop('UnitTestsCollection');
+      c = db._create('UnitTestsCollection');
+    },
+
+    ////////////////////////////////////////////////////////////////////////////////
+    /// @brief tear down
+    ////////////////////////////////////////////////////////////////////////////////
+
+    tearDown: function () {
+      // we need try...catch here because at least one test drops the collection itself!
+      try {
+        c.unload();
+        c.drop();
+      } catch (err) {
+      }
+      c = null;
+      if (view) {
+        view.drop();
+        view = null;
+      }
+      internal.wait(0.0);
+    },
+
+    ////////////////////////////////////////////////////////////////////////////
+    /// @brief should honor rollbacks of inserts
+    ////////////////////////////////////////////////////////////////////////////
+    testRollbackInsertWithLinks1 : function () {
+
+      let meta = { links: { 'UnitTestsCollection' : { fields: {text: {analyzers: [ "myText" ] } } } } };
+      view = db._createView("UnitTestsView", "arangosearch", {});
+      view.properties(meta);
+      let links = view.properties().links;
+      assertNotEqual(links['UnitTestsCollection'], undefined);
+
+      c.save({ _key: "full", text: "the quick brown fox jumps over the lazy dog" });
+      c.save({ _key: "half", text: "quick fox over lazy" });
+
+      try {
+        db._executeTransaction({
+          collections: {write: 'UnitTestsCollection'},
+          action: function() {
+            const db = require('internal').db;
+  
+            c.save({ _key: "other_half", text: "the brown jumps the dog" });
+            c.save({ _key: "quarter", text: "quick over" });
+  
+            /*let result = db._query(qqWithSync).toArray();
+            assertEqual(result.length, 4);
+            assertEqual(result[0]._key, 'half');
+            assertEqual(result[1]._key, 'quarter');
+            assertEqual(result[2]._key, 'other_half');
+            assertEqual(result[3]._key, 'full');*/
+            throw "myerror";
+          }
+        });
+      } catch (err) {
```

Comment:
we should definitely add a `fail()` before the catch, otherwise the code inside the catch block may not be executed.